src/AstSemanticChecker.cpp

```diff
@@ -293,8 +293,8 @@ void AstSemanticChecker::checkProgram(AstTranslationUnit& translationUnit) {
         }
 
         // Check argument types.
-        for (size_t i = 0; i < fun.getArity(); i++) {
-            const AstArgument* arg = fun.getArg(i);
+        for (size_t i = 0; i < fun.getArguments().size(); i++) {
```

Comment:
can you change this loop to size_t i=0; for(auto arg: fun.getArguments()) { ....; i++}